integrationtests/tests/tests.rs

```diff
@@ -386,9 +386,12 @@ async fn receive_lightning_payment_valid_preimage() {
     .unwrap();
 
     // Gateway deposits ecash to trigger preimage decryption by the federation
+    // It can deposit more ecash than is required,
+    // but only the right amount will be spent towards buying preimage offer
+    let deposit_amount = payment_amount + sats(100);
     let (outpoint, contract_id) = gateway
         .server
-        .buy_preimage_offer(invoice.invoice.payment_hash(), &payment_amount, rng())
+        .buy_preimage_offer(invoice.invoice.payment_hash(), &deposit_amount, rng())
```

Comment:
Okay, I made up a test scenario where gateway proposes to buy preimage offer with an amount > the correct invoice amount, and it still doesn't lose funds because `buy_preimage_offer` performs the correct logic to return change to the gateway...

Reply:
I should make it clear the gateway should use the right `invoice_amount` in normal operations. This is a contrived scenario to reproduce the bug and make sure gw client does the right thing :)